src/base.jl

```diff
@@ -307,3 +307,13 @@ function Base.:*(K::AbstractKroneckerProduct, a::Number)
     A, B = getmatrices(K)
     kronecker(A, B * a)
 end
+
+# SOLVING
+using LinearAlgebra: checksquare
+function LinearAlgebra.:\(K::AbstractKroneckerProduct{T}, c::AbstractVector{T}) where {T}
+    !issquare(K.A) && throw(DimensionMismatch("matrix A is not square: dimensions are " * size(K.A)))
```

Comment:
Perhaps this should fallback to a slow method rather than bailing out entirely if a non-square matrix is encountered?

Reply:
You're right, I wasn't thinking of least-squares. I'll make the change.

Reply:
I just updated it. Let me know if anything else comes up.